src/pages/TokenDetails/index.tsx

```diff
@@ -14,38 +14,41 @@ export const pageTimePeriodAtom = atomWithStorage<TimePeriod>('tokenDetailsTimeP
 export default function TokenDetailsPage() {
   const { tokenAddress, chainName } = useParams<{ tokenAddress?: string; chainName?: string }>()
   const chain = validateUrlChainParam(chainName)
-  const pageChainId = CHAIN_NAME_TO_CHAIN_ID[chain]
   const isNative = tokenAddress === NATIVE_CHAIN_ID
   const [timePeriod, setTimePeriod] = useAtom(pageTimePeriodAtom)
-  const [contract, duration] = useMemo(
-    () => [
-      { address: isNative ? nativeOnChain(pageChainId).wrapped.address : tokenAddress ?? '', chain },
-      toHistoryDuration(timePeriod),
-    ],
-    [chain, isNative, pageChainId, timePeriod, tokenAddress]
+  const [address, duration] = useMemo(
+    () => [isNative ? getNativeTokenDBAddress(chain) : tokenAddress ?? '', toHistoryDuration(timePeriod)],
+    [chain, isNative, timePeriod, tokenAddress]
   )
 
-  const { data: tokenQuery, loading: tokenQueryLoading } = useTokenQuery({
+  const { data: tokenQuery } = useTokenQuery({
     variables: {
-      contract: isNative ? { address: getNativeTokenDBAddress(chain), chain } : contract,
+      address,
+      chain,
     },
   })
 
   const { data: tokenPriceQuery } = useTokenPriceQuery({
     variables: {
-      contract,
+      address,
+      chain,
       duration,
     },
   })
 
-  if (!tokenQuery || tokenQueryLoading) return <TokenDetailsPageSkeleton />
+  const [currentPriceQuery, setCurrentPriceQuery] = useState(tokenPriceQuery)
+  useEffect(() => {
```

Comment:
can you write a comment above here that explains why this state + useEffect is necessary? took me a bit of parsing to understand, but is it basically like `tokenPriceQuery` can sometimes be null, so use state to make sure it's always defined?